File: src/main/java/textGame/Inventory.java
package textGame;

public class Inventory {
    private Item[] items = new Item[3];
    public Item[] getItems() {
        return items;
    }
    public Item getItemByName(String name) {
        for (Item item : items) {
            if (item != null && item.getName().equals(name)) {
                return item;
            }
        }
        return null;
    }
    public boolean addItem(Item item) {
        for (int i = 0; i < items.length; i++) {
            if (items[i] == null) {
                items[i] = item;
                return true;
            }
        }
        return false;
    }
    public boolean removeItem(Item item) {
        for (int i = 0; i < items.length; i++) {
            if (items[i] == item) {
                items[i] = null;
                return true;
            }
        }
        return false;
    }
    public boolean isFull() {
        boolean isFull = true;
        for (Item item : items) {
            if (item == null) {
                isFull = false;
                break;
            }
        }
        return isFull;
    }
    public String toString() {
        String output = "( ";
        for (int i = 0; i < items.length; i++) {
            output = output + items[i] + " ";
        }
        output = output + ")";
        return output;
    }
}

File: src/main/java/textGame/Inventory.kt
package textGame

class Inventory (size: Int) {
    val items: Array<Item?> = arrayOfNulls(size)
    fun getItemByName(name: String): Item? {
        return items.filterNotNull().find{it.getName() == name}
    }

    fun addItem(item: Item?): Boolean {
        for (i in items.indices) {
            if (items[i] == null) {
                items[i] = item
                return true
            }
        }
        return false
    }

    fun removeItem(item: Item): Boolean {
        for (i in items.indices) {
            if (items[i] === item) {
                items[i] = null
                return true
            }
        }
        return false
    }

    val isFull: Boolean
        get() {
            var isFull = true
            for (item in items) {
                if (item == null) {
                    isFull = false
                    break
                }
            }
            return isFull
        }

    override fun toString(): String {
        var output = "( "
        for (i in items.indices) {
            output = output + items[i] + " "
        }
        output = "$output)"
        return output
    }
}